src/librustc_typeck/collect.rs

```diff
@@ -2634,6 +2636,16 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
     codegen_fn_attrs
 }
 
+fn inline_exportable(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
+    // Functions marked with #[inline] are only ever codegened
+    // with "internal" linkage and are never exported unless we're
+    // building a `staticlib` or `cdylib` and they are marked
+    // `#[no_mangle]`.
+    tcx.codegen_fn_attrs(def_id).flags.contains(CodegenFnAttrFlags::NO_MANGLE)
```

Comment:
There is also `#[export_name = "..."]` and `#[linkage = "..."]` to consider so this should really be calling [`contains_extern_indicator`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/middle/codegen_fn_attrs/struct.CodegenFnAttrs.html#method.contains_extern_indicator).

Reply:
I wasn't aware of `contains_extern_indicator`, thank you so much for the suggestion!